stackslib/src/net/dns.rs

```diff
@@ -423,16 +416,18 @@ mod test {
             client.try_recv().unwrap();
 
             for name in names.iter() {
-                if resolved_addrs.contains_key(&name.to_string()) {
-                    continue;
-                }
-                match client.poll_lookup(name, 80).unwrap() {
-                    Some(addrs) => {
-                        test_debug!("name {} addrs: {:?}", name, &addrs);
-                        resolved_addrs.insert(name.to_string(), addrs);
-                        break;
+                match resolved_addrs.entry(name.to_string()) {
+                    Entry::Occupied(_) => {
+                        continue;
```

Comment:
`continue` isn't necessary because we're at the end of the loop. How about changing this block to an `if let Entry::Vacant(e)...`